### PR TITLE
OGM-1420 Test ElementCollection is loaded from JPQL query

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1420

### CODE REVIEW

**Comment by DavideD** [comment]
THis is a small testcase I had for a while to cover a usecase we don't conver right now

File: core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[]{ FoodsCosmeticsMedicines.class };
+	}
+
+	@Entity(name = "FoodsCosmeticsMedicines")
+	@Indexed
+	@Table(name = "foodsCosmeticsMedicines")
+	public static class FoodsCosmeticsMedicines implements Serializable {
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java]: Is it possible to share the same test Entities for both test cases?

File: core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+			session.delete( session.load( FoodsCosmeticsMedicines.class, medicine.getName() ) );
+		} );
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[]{ FoodsCosmeticsMedicines.class };
+	}
+
+	@Entity(name = "FoodsCosmeticsMedicines")
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java]: the attribute name is it necessary?

File: core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryWithJPATest.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+		inTransaction( em -> {
+			@SuppressWarnings("unchecked")
+			List<FoodsCosmeticsMedicines> loaded = em.createQuery( "FROM FoodsCosmeticsMedicines" ).getResultList();
+
+			assertThat( loaded ).containsOnly( medicine );
+			assertThat( loaded.get( 0 ).getQtyContentsList() ).containsOnly( content1, content2 );
+		} );
+
+		inTransaction( em -> {
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryWithJPATest.java]: Maybe we can use the new base method: deleteAll. To clean up the datastore.

**Comment by DavideD** [comment]
Thanks @fax4ever

**Pull request closed without merging**